### ISSUE TITLE
what's different with vite

### ISSUE THREAD

**Issue by joebnb** [HairyRabbit/esm-bundle#3]
it seems vite do same thing like esm-bundle but in different step,if we in development the dependencie tree may change high frequently we manage version by semantic version control.so the node_modules version may change in each install if we want keep cahe up to date we need run webpack build packages in each install.i'm not sure did this worth for develope

**Comment by HairyRabbit** [comment]
yes, this package helpful to implement your web develop toolchains like vite. you may not need this when you use webpack under development mode. it just design for browser native esm usage. in my case, I use this tool as a server middleware. like the code: `import { useState } from 'react'` .when the browser request `react`, this tool will bundle it before not cache. 

mostly you may not care about dependency sources, just known the exports(default and named) and run your code without error enough. here is some details. at first, this tool not build common chunks, e.g `react` and `react-dom` both depends on the `object-assign` library. vite will build it as common chunk, but this tool not, it will bundle the library into their bundles; and next, not all of bundled dependencis can run without errors. I test `rollup`, `webpack`, `esbuild` to bundle `react/jsx-runtime`, only `rollup` works fine for me, because the re-exports a dependency is diffcult to analysed.

usually the dependencies versions locked by package manager when you installed successful. you only need rebundle when upgrade the dependency version, like update react@17.0.2 to react@18.0.0.

webpack is not in my development toolchains, but I use it to bundle production code.
